datalad/crawler/nodes/annex.py

```diff
@@ -336,7 +336,7 @@ def __init__(self, path=None,
         self._providers = Providers.from_config_files()
         self.yield_non_updated = yield_non_updated
 
-        if (not allow_dirty) and self.repo.dirty:
+        if (not allow_dirty) and self.repo.dirty():
```

Comment:
I think I have expressed it before but could reiterate -- I see no reason or beauty in making `dirty` into a method whenever all the other similar methods are called with `is_` prefix within our Repo classes:
```shell
(git)hopa:~datalad/datalad[pr-1294]git
$> git grep -A1 '@property' -- datalad/support/*repo.py     
datalad/support/annexrepo.py:    @property
datalad/support/annexrepo.py-    def dirty(self):
--
datalad/support/annexrepo.py:    @property
datalad/support/annexrepo.py-    def default_backends(self):
--
datalad/support/annexrepo.py:    @property
datalad/support/annexrepo.py-    def uuid(self):
--
datalad/support/gitrepo.py:    @property
datalad/support/gitrepo.py-    def config(self):
--
datalad/support/gitrepo.py:    @property
datalad/support/gitrepo.py-    def dirty(self):
--
datalad/support/gitrepo.py:    @property
datalad/support/gitrepo.py-    def count_objects(self):
changes on filesystem:                                                                                                                                  
 datalad/distribution/publish.py  | 23 ++++++++++++++++++++++-
 datalad/interface/common_opts.py |  5 +++++
(dev) 2 12000.....................................:Tue 14 Feb 2017 01:55:22 PM EST:.
(git)hopa:~datalad/datalad[pr-1294]git
$> git grep -B1 'def is_' -- datalad/support/*repo.py  
datalad/support/annexrepo.py-    @classmethod
datalad/support/annexrepo.py:    def is_valid_repo(cls, path, allow_noninitialized=False):
--
datalad/support/annexrepo.py-
datalad/support/annexrepo.py:    def is_direct_mode(self):
--
datalad/support/annexrepo.py-
datalad/support/annexrepo.py:    def is_crippled_fs(self):
--
datalad/support/annexrepo.py-    @normalize_paths
datalad/support/annexrepo.py:    def is_under_annex(self, files, allow_quick=True, batch=False):
--
datalad/support/annexrepo.py-    @normalize_paths(serialize=True)
datalad/support/annexrepo.py:    def is_available(self, file_, remote=None, key=False, batch=False):
--
datalad/support/gitrepo.py-    @classmethod
datalad/support/gitrepo.py:    def is_valid_repo(cls, path):
--
datalad/support/gitrepo.py-
datalad/support/gitrepo.py:    def is_with_annex(self, only_remote=False):
```
and then I do not see why not to leave `dirty` to be a property as a lightweight shim to `is_dirty` call, thus making this PR easier to review (now full of `.dirty` -> `.dirty()`) and code which possibly uses it already outside not needing to change